clients/src/main/java/org/apache/kafka/common/requests/DescribeConfigsResponse.java

```diff
@@ -179,7 +179,8 @@ public boolean isReadOnly() {
         DYNAMIC_BROKER_CONFIG((byte) 2),
         DYNAMIC_DEFAULT_BROKER_CONFIG((byte) 3),
         STATIC_BROKER_CONFIG((byte) 4),
-        DEFAULT_CONFIG((byte) 5);
+        DEFAULT_CONFIG((byte) 5),
+        DYNAMIC_BROKER_LOGGER_CONFIG((byte) 6);
```

Comment:
Agreed. I thought it was ordered by ID. We need to keep DEFAULT_CONFIG=5 for old clients, right?

Reply:
Ok I learned through test failures that that's not the ID but the ordinal. We should change them and code shouldn't depend on those anyway

Reply:
Wait, now I'm confused. I re-checked the code based on your comment and it does look like those numbers are the IDs that we are returning to clients, so changing the numbers break our protocol.  (looking at the `forId` method).
What did your tests reveal? Where does the actual ID come from?

Also maybe a thing to keep in mind (and document): It looks like older adminclient talking to a new broker may see the config source as "UNKNOWN CONFIG". I don't think it is a big deal, but just something to be aware of.

Reply:
You're 100% right. This ID is actually an index.

The reason my test failed was because I had the code like this:
```
public enum ConfigSource {
        UNKNOWN_CONFIG((byte) 0),
        TOPIC_CONFIG((byte) 1),
        DYNAMIC_BROKER_CONFIG((byte) 2),
        DYNAMIC_DEFAULT_BROKER_CONFIG((byte) 3),
        STATIC_BROKER_CONFIG((byte) 4),
        DYNAMIC_BROKER_LOGGER_CONFIG((byte) 6), -- NEW
        DEFAULT_CONFIG((byte) 5);
```
in order to keep the convention of defaults at the bottom. The test would fail because it would get a byte ID of 6 and get values[6], which is DEFAULT_CONFIG and not the expected dynamic broker logger.

So it turns out we can't keep the convention here :(

Reply:
An old AdminClient talking to a new broker should never see this config source at all because it would never request the broker loggers configs